src/modules/drawing/components/AddVertexButtonOverlay.vue

```diff
@@ -0,0 +1,107 @@
+<script setup>
+import { LineString } from 'ol/geom'
+import Overlay from 'ol/Overlay'
+import { onMounted, ref, watch } from 'vue'
+import { inject } from 'vue'
+import { onUnmounted } from 'vue'
+
+import AddVertexButton from './AddVertexButton.vue'
+
+const props = defineProps({
+    lineString: {
+        type: LineString,
+        required: true,
+    },
+})
+
+const olMap = inject('olMap')
+
+const firstButtonOverlay = ref(null)
+const lastButtonOverlay = ref(null)
+
+const firstButtonCoordinate = ref(null)
+const lastButtonCoordinate = ref(null)
+
+const calculateOffset = (point1, point2, distance = 35) => {
+    if (!point1 || !point2) {
+        return [distance, -distance]
+    }
+
+    // Vector from point1 to point2
+    const dx = point2[0] - point1[0]
+    const dy = point2[1] - point1[1]
+
+    // Normalize the vector
+    const length = Math.sqrt(dx * dx + dy * dy)
+    if (length === 0) return [distance, -distance]
+
+    // Get unit vector in opposite direction
+    const ux = -dx / length
+    const uy = -dy / length
+
+    // There is minus in y-direction because the y-axis is inverted in the map
+    return [ux * distance, -uy * distance]
+}
+
+const updateButtonPositions = () => {
+    const coordinates = props.lineString.coordinates
```

Comment:
and if you switch to a computed you can replace the use here by a use of the composed too